src/hooks/useFauxCaretCssVars.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react'
+import Path from '../@types/Path'
+import { isMobileSafari } from '../browser'
+import { isEndOfElementNode, isNote, isStartOfElementNode } from '../device/selection'
+import editingValueStore from '../stores/editingValue'
+
+/** Returns CSS variables that will suppress faux carets at the start or end of thoughts or notes.
+ * */
+const useFauxCaretCssVars = (
+  editing: boolean | null,
+  // Undo followed by redo causes the component to re-mount with the same prop values,
+  // but fadeThoughRef is re-created and will trigger an update.
+  fadeThoughtElement: HTMLDivElement | null,
+  isCursor: boolean,
+  isTableCol1: boolean,
+  path: Path,
+) => {
+  const [showLineEndFauxCaret, setShowLineEndFauxCaret] = useState(false)
+  const [showLineStartFauxCaret, setShowLineStartFauxCaret] = useState(false)
+  const [showNoteLineStartFauxCaret, setShowNoteLineStartFauxCaret] = useState(false)
+  const [showNoteLineEndFauxCaret, setShowNoteLineEndFauxCaret] = useState(false)
+
+  // Hide the faux caret when typing occurs.
+  editingValueStore.useEffect(() => {
+    if (!isMobileSafari()) return
+    setShowLineStartFauxCaret(false)
+    setShowLineEndFauxCaret(false)
+    setShowNoteLineEndFauxCaret(false)
+  })
+
+  // If the thought isCursor and edit mode is on, position the faux cursor at the point where the
+  // selection is created.
+  useEffect(() => {
+    if (!isMobileSafari()) return
+    if (editing && isCursor) {
+      // The selection ranges aren't updated until the end of the frame when the thought is focused.
+      setTimeout(() => {
+        if (editing && isCursor) {
+          if (isNote()) {
```

Comment:
You should be able to use `store.getState().noteFocus` since we're already tracking the note state in the Redux store.

Either approach is reasonable, but less dependence on the DOM is usually preferred. Using `state.noteFocus` will eliminate the direct dependency between `selection` and the Note's aria-label.